src/core/socket.c

```diff
@@ -1224,6 +1224,7 @@ static int usbffs_dispatch_eps(SocketPort *p) {
 
 static int socket_open_fds(Socket *s) {
         _cleanup_(mac_selinux_freep) char *label = NULL;
+        _cleanup_free_ const char *ep;
```

Comment:
Also, please *always* initialize variables annotated with `_cleanup_free_` to `NULL`, in order to address the issue I described.